### PR TITLE
chore(poetry): update python to 3.11

### PR DESCRIPTION
**Pull request by IamAbbey**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ line_length = 120
 [tool.coverage.run]
 branch = true
 source = ["."]
-omit = ["**/tests/**.py"]
+omit = ["**/tests/*.py"]
```
Comment (marns93) [inline pyproject.toml:37]: In other repositories we are using `omit = ["**/tests/**"]`, for instance: https://github.com/moneymeets/youtrack-sdk/blob/92106b45ffcaae8eb967a742782ae31414bbae61/pyproject.toml#L38

**Review by marns93 — CHANGES_REQUESTED** [review]

**Review by felix11h — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (IamAbbey) [reply pyproject.toml:37]: For updated version of coverage 
omit = ["**/tests/**"] will fail (not work) I can confirm that

There was major changes to file pattern matching
reference: https://coverage.readthedocs.io/en/stable/changes.html#version-7-0-0b1-2022-12-03

Comment (marns93) [reply pyproject.toml:37]: Okay, so you mean we need to update https://github.com/moneymeets/youtrack-sdk/blob/master/pyproject.toml#L38 also? This is already merged with latest version.

**Comment by marns93** [comment]
@IamAbbey Please update https://github.com/moneymeets/action-s3-artifact/blob/master/action.yml#L24 as well.

**Review by marns93 — APPROVED** [review]

**Pull request merged**